src/main/java/com/icthh/xm/ms/dashboard/repository/impl/permitted/UiDataPermittedRepository.java

```diff
@@ -0,0 +1,46 @@
+package com.icthh.xm.ms.dashboard.repository.impl.permitted;
+
+import com.icthh.xm.commons.permission.repository.PermittedRepository;
+import com.icthh.xm.commons.permission.service.PermissionCheckService;
+import com.icthh.xm.ms.dashboard.domain.UiData;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+public class UiDataPermittedRepository extends PermittedRepository {
+    public UiDataPermittedRepository(PermissionCheckService permissionCheckService) {
+        super(permissionCheckService);
+    }
+
+    public Page<UiData> findAllByTypeKeyAndOwner(String typeKey, String key, String owner, Pageable pageable, String privilegeKey) {
+        String typeKeyCondition = "typeKey = :typeKey";
+        String ownerCondition = "owner = :owner";
+        String keyCondition = "key = :key";
+
```

Comment:
could this consts exist out of the method to avoid it's recreation several times at runtime?